### DISCUSSION TITLE
Pull requests don't get environment variables

### DISCUSSION THREAD

**Discussion by JoyShaheb** [coollabsio/coolify#3612] [Q&A]
I have noticed something is that, after deploying the main branch, when i deploy the Pull requests before merging to main branch, the env variables are not shared with the PR deployments. am i doing something wrong or is coolify missing something ? please advise 🙏

**Comment by xadacka** [comment]
I, too, would like to know how to resolve this!

**Comment by KianAnbarestani** [comment]
This is expected behavior in Coolify—environment variables are not automatically shared between the main branch and PR deployments. Each deployment (main, PR, preview) is treated as an isolated environment for security and separation of concerns.

If you want PR deployments to access certain env variables, you need to:

- Define them explicitly in the PR deployment config.
- Or use global environment variables (under Project > Environment Variables) if they should apply across all branches.

So no, you're not doing anything wrong—Coolify just doesn't inherit `main` branch envs into PRs by default.
